go/worker/storage/committee/node.go

```diff
@@ -388,6 +394,29 @@ func (n *Node) getMetricLabels() prometheus.Labels {
 	}
 }
 
+func (n *Node) WaitForRound(round uint64, root *storageApi.Root) (<-chan uint64, error) {
+	retCh := make(chan uint64, 1)
+
+	if root != nil {
+		round = root.Version
+	}
+
+	n.syncedLock.RLock()
```

Comment:
This probably needs to be upgraded to a write lock as you are modifying `roundWaiters` below.